### ISSUE TITLE
Customizable evaluation results filters

### ISSUE THREAD

**Issue by janno42** [e-valuation/EvaP#1786]
Currently, an evaluation's results page can be viewed as "Public" (shows results that are visible for everyone logged in), "Own" (shows all results that are visible for the current user), and "Export" (shows all results for general questions and the results for the current user).

This should be changed to two customizable filters which can be selected via switch button groups (two button groups placed below each other in the top right of the page):
- General results (only visible if current user can see text answers for general questions (reviewers, responsibles of this evaluation, contributors in this evaluation where `TextAnswerVisibility` is set to `GENERAL_TEXTANSWERS`, and delegates of such responsibles and contributors))
    - Show text answers
    - Hide text answers
- Contributor results (only visible if current user is reviewer, contributor in this evaluation, or delegate of such a contributor)
    - Show text answers
    - Hide text answers
    - Only my results (only visible if current user is contributor in this evaluation and for staff users who opened this view via the contributor's profile like in the current "Export" view)

**Comment by richardebeling** [comment]
@Tillbtn @ybrnr @JosefValentin2004 What's your status here, do you want to continue working on this or can we unassign/reassign?

**Comment by JosefValentin2004** [comment]
@richardebeling No, I wanna continue. We are almost done anyways but Prüfungsphase :(

**Comment by jooooosef** [comment]
We now have the changed buttons, with:
general: full (with textanswers), ratings (without textanswers) for the general results
contributor: full (all text answers from all contributors), ratings (no text answers from any contributors) and personal (show ALL textanswers regarding that contributor).

Should personal also include the so called private textanswers (regarding the contributor)? Before when there was only one set options with public, full and export, in export you couldnt see the private text answers.

**Comment by janno42** [comment]
> Should personal also include the so called private textanswers (regarding the contributor)? Before when there was only one set options with public, full and export, in export you couldnt see the private text answers.

Private text answers should be shown when selecting "personal" - the respective tests need to be changed.

**Comment by richardebeling** [comment]
@jooooosef Can you push your current state to some branch and create a pull request for that (possibly marked WIP if you want) so we can take a first look at the code and especially the tests?

**Comment by jooooosef** [comment]
@richardebeling I just did, marked as draft. I started to change the tests a little but only did it for students so far.
